packages/mattermost-redux/src/actions/threads.ts

```diff
@@ -268,20 +260,46 @@ export function markAllThreadsInTeamRead(userId: string, teamId: string) {
     };
 }
 
-export function markThreadAsUnread(userId: string, teamId: string, threadId: string, postId: string) {
+export function markThreadAsUnread(userId: string, teamId: string, threadId: string) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        try {
-            await Client4.markThreadAsUnreadForUser(userId, teamId, threadId, postId);
-        } catch (error) {
-            forceLogoutIfNecessary(error, dispatch, getState);
-            dispatch(logError(error));
-            return {error};
+        const getPostsForThread = makeGetPostsForThread();
+        let posts = getPostsForThread(getState(), threadId);
+
+        // load posts in thread if they are not loaded already
+        if (posts.length < 2) {
```

Comment:
we could make the check more safe by checking the expected number of replies, vs the actual stored replies in redux.

e.g

```ts
if (thread.replyCount !== posts.length - 1) {
```
something like that

Reply:
Thanks for the recommendation, updated